_books/ion-1-1/src/macros/macros_by_example.md

```diff
@@ -262,31 +243,41 @@ single string:
 ```ion
 (:make_string)                 ⇒ ""
 (:make_string "a")             ⇒ "a"
-(:make_string "a" "b"    )     ⇒ "ab"
+(:make_string "a" "b")         ⇒ "ab"
 (:make_string "a" "b" "c")     ⇒ "abc"
 (:make_string "a" "b" "c" "d") ⇒ "abcd"
 ```
 
-To make this work, the definition of make_string is effectively:
+To make this work, the declaration of `make_string` is effectively:
 
 ```ion
-(macro make_string (parts*) …)
+(macro make_string (parts*) /*...*/)
 ```
 
-This says that `parts` is comprised of zero or more argument expressions.
-The `*` cardinality modifier, when applied to a final parameter, declares that
-“all the rest” of the arguments will be passed to that one name.
+The `*` is a _[cardinality](#cardinality)_ modifier.
+A parameter's cardinality dictates both the number of argument expressions it can accept and the number of values its expansion can produce.
 
-> [!NOTE]
-> The Ion grammar treats identifiers like `text` and operators like `*` as separate tokens
-> regardless of whether they are separated by whitespace. We think it's easier to read without
-> whitespace and will use that convention from now on.
+In the examples so far, all parameters have had a cardinality of `exactly-one`, which is the default.
+The `parts` parameter has a cardinality of `zero-or-more`, meaning:
+1. It can accept `zero-or-more` argument expressions.
+2. When expanded, it will produce `zero-or-more` values.
 
-At this point our distinction between parameters and arguments becomes apparent, since
+When the final parameter in the macro signature is `zero-or-more`, "all of the rest" of the argument expressions will be passed to that parameter.
```

Comment:
Does it work with one-or-more too?

Reply:
It does, but `one-or-more` hasn't been introduced yet.